### PR TITLE
[sdk-k8s#207] Stop reading closed channel

### LINKED ISSUE #207: ETCD registry watch starts receiving tons of events
## Place in code
https://github.com/networkservicemesh/sdk-k8s/blob/6a4dea64961fcdb8dee6b5023d7259106907bcfd/pkg/registry/etcd/nse_server.go#L122-L132

## Expected behavior
This code should receive updates only on NSE updates (Register, Unregister).

## Current behavior
After some time working (30-40 min) it starts receiving tons of events with `event.Object == nil`. It results in looping on `continue` and using huge amount of CPU resources.

### PR DESCRIPTION
**Pull request by Bolodya1997**
## Description
Restarts NS, NSE watchers on result channel close.

## Issue link
Closes #207  

## How Has This Been Tested?

- [ ] Added unit testing to cover
- [x] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [x] Bug fix
- [ ] New functionallity
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

**Review by denis-tingaikin — COMMENTED** [review]
Looks good.

Question: Can we with this patch remove this line? https://github.com/networkservicemesh/sdk-k8s/blob/main/pkg/registry/chains/registryk8s/registry-k8s_test.go#L41

**Comment by Bolodya1997** [comment]
> Looks good.
> 
> Question: Can we with this patch remove this line? https://github.com/networkservicemesh/sdk-k8s/blob/main/pkg/registry/chains/registryk8s/registry-k8s_test.go#L41

No, it doesn't relate anyhow to this.

File: pkg/registry/etcd/ns_server.go [hunk 1]
```diff
@@ -73,24 +74,34 @@ func (n *etcdNSRegistryServer) Register(ctx context.Context, request *registry.N
...
-			model := event.Object.(*v1.NetworkService)
-			item := (*registry.NetworkService)(&model.Spec)
-			if matchutils.MatchNetworkServices(query.NetworkService, item) {
-				err := s.Send(item)
-				if err != nil {
-					return err
+		watcher, err := n.client.NetworkservicemeshV1().NetworkServices(n.ns).Watch(s.Context(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
```
Comment (denis-tingaikin) [inline pkg/registry/etcd/ns_server.go:83]: Why do we need to re-create watcher?

Comment (Bolodya1997) [reply pkg/registry/etcd/ns_server.go:83]: We are still processing watching Find, so we still need a watcher.

Comment (denis-tingaikin) [reply pkg/registry/etcd/ns_server.go:83]: As you internally pointed out, we don't actually have the option to use a perpetual observer https://github.com/kubernetes/apiserver/blob/master/pkg/endpoints/handlers/get.go#L258

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**